src/storage/accessors/FileDataAccessor.ts

```diff
@@ -277,63 +280,50 @@ export class FileDataAccessor implements DataAccessor {
-    const quads: Quad[] = [];
-    const childURIs: string[] = [];
-    const files = await fsPromises.readdir(link.filePath);
+  private async* getChildMetadata(link: ResourceLink): AsyncIterableIterator<RepresentationMetadata> {
+    const dir = await fsPromises.opendir(link.filePath);
 
     // For every child in the container we want to generate specific metadata
-    for (const childName of files) {
-      // Hide metadata files from containment triples
+    for await (const dirent of dir) {
+      const childName = dirent.name;
+      // Hide metadata files
       if (this.isMetadataPath(childName)) {
         continue;
       }
 
       // Ignore non-file/directory entries in the folder
-      const childStats = await fsPromises.lstat(joinFilePath(link.filePath, childName));
-      if (!childStats.isFile() && !childStats.isDirectory()) {
+      if (!dirent.isFile() && !dirent.isDirectory()) {
         continue;
       }
 
       // Generate the URI corresponding to the child resource
       const childLink = await this.resourceMapper
-        .mapFilePathToUrl(joinFilePath(link.filePath, childName), childStats.isDirectory());
+        .mapFilePathToUrl(joinFilePath(link.filePath, childName), dirent.isDirectory());
 
       // Generate metadata of this specific child
-      const subject = DataFactory.namedNode(childLink.identifier.path);
-      quads.push(...generateResourceQuads(subject, childStats.isDirectory()));
-      quads.push(...this.generatePosixQuads(subject, childStats));
-      childURIs.push(childLink.identifier.path);
+      const childStats = await fsPromises.lstat(joinFilePath(link.filePath, childName));
+      const metadata = new RepresentationMetadata(childLink.identifier);
+      addResourceMetadata(metadata, childStats.isDirectory());
+      this.addPosixMetadata(metadata, childStats);
+      yield metadata;
     }
-
-    // Generate containment metadata
-    const containsQuads = generateContainmentQuads(
-      DataFactory.namedNode(link.identifier.path), childURIs,
-    );
-
-    return quads.concat(containsQuads);
   }
 
   /**
    * Helper function to add file system related metadata.
-   * @param subject - Subject for the new quads.
+   * @param metadata - metadata object to add to
    * @param stats - Stats of the file/directory corresponding to the resource.
    */
-  private generatePosixQuads(subject: NamedNode, stats: Stats): Quad[] {
-    const quads: Quad[] = [];
-    pushQuad(quads, subject, DC.terms.modified, toLiteral(stats.mtime.toISOString(), XSD.terms.dateTime));
-    pushQuad(quads, subject, POSIX.terms.mtime, toLiteral(
-      Math.floor(stats.mtime.getTime() / 1000), XSD.terms.integer,
-    ));
+  private addPosixMetadata(metadata: RepresentationMetadata, stats: Stats): void {
+    metadata.add(DC.terms.modified, toLiteral(stats.mtime.toISOString(), XSD.terms.dateTime));
+    metadata.add(POSIX.terms.mtime, toLiteral(Math.floor(stats.mtime.getTime() / 1000), XSD.terms.integer));
```

Comment:
Wouldn't this belong in a utility or base class? It doesn't seem specific to files.

Reply:
True, we could also set these for all the other data accessors. Will create an issue for this.

Reply:
No wait, just realized this is file specific since it uses the Stats object. We could have a utility function that takes the time value as input and then adds this, but we only have the file system that supports that currently. The `posix` namespace is specifically for files as well I think.

Reply:
👍